### ISSUE TITLE
Triplet loss for training

### ISSUE THREAD

**Issue by agusgun** [rcmalli/keras-vggface#50]
Can I know if you use triplet based loss to train the model or not for the VGG16 and RESNET50 model?

Thank you very much

**Comment by rcmalli** [comment]
It should be possible since the learning method is not model dependent.

**Comment by TuanAnhNguyen14111998** [comment]
@rcmalli  Can you guide me fine tuning VGGFace Resnet50 with triplet loss with keras? I'm really having trouble doing this! val_loss remains unchanged, always stays at 1,000, although loss has decreased.

**Comment by sherlockchou86** [comment]
1.load pre-trained weights without top layers, as base model
2.add 3 inputs on base model to create a new model, outputs are feature vectors for 3 inputs, which will be used later in loss function
3.define your triplet loss functions Max(d1-d2_margin, 0), d1 is the distance between anchor features and positive features, d2 is the distance between anchor features and negative features
4.train your own datasets on the new model, 3 inputs each time and no labels needed
